src/assets.rs

```diff
@@ -207,9 +207,23 @@ impl HighlightingAssets {
                     None => ext_syntax.or(line_syntax),
                 }
             }
-            (None, InputFile::StdIn) => String::from_utf8(reader.first_line.clone())
+            (None, InputFile::StdIn(None)) => String::from_utf8(reader.first_line.clone())
                 .ok()
                 .and_then(|l| self.syntax_set.find_syntax_by_first_line(&l)),
+            (None, InputFile::StdIn(Some(file_name))) => self
+                .syntax_set
+                .find_syntax_by_extension(file_name.to_str().unwrap())
+                .or_else(|| {
+                    self.syntax_set.find_syntax_by_extension(
+                        Path::new(file_name)
+                            .extension()
+                            .and_then(|x| x.to_str())
+                            .unwrap_or(""),
+                    )
+                })
+                .or(String::from_utf8(reader.first_line.clone())
+                    .ok()
+                    .and_then(|l| self.syntax_set.find_syntax_by_first_line(&l))),
```

Comment:
I don't like the fact that the whole logic from the `(None, InputFile::Ordinary(ofile))` case above is repeated here. We should extract this into a separate method or try to move both cases into the same match arm.

Reply:
Yep, that makes sense. I chose the helper method approach.